src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,152 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
-
+        if (threshold() == size) {
+            resize();
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        linkToBucket(table, newNode);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
+            return null;
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        int position = hash % table.length;
+        position = validatePosition(position, table);
+        Node<K, V> current = new Node<>(hash, key, null, null);
+        return findNode(position, current);
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] newTable = new Node[table.length * GROW_FACTOR];
+        size = 0;
+        copyElements(table, newTable);
+        table = newTable;
+    }
```

Comment:
There is a much simpler approach 😉 
* save old table to the var
* reassign a new table with a new length to the `table` var
* drop `size` var to 0;
* for loop through old buckets
  * while loop through nodes and call `put(K key, V value)` on each of the node

clean unused methods after that